### ISSUE TITLE
[flow] Error parsing Flow $ObjMap

### ISSUE THREAD

**Issue by michalkvasnicak** [babel/babel-eslint#466]
I don't know if this issue should be file here or somewhere else, so I'm sorry it this repository does not have anything to do with this issue.

Packages:

* `eslint 3.19.0`
* `babel-eslint 7.2.3`

Given following code:

```js
// @flow

type Test = {
  [key: string]: () => () => any,
}

type Props<T: Test> = {
  prop: T,
  render(a: $ObjMap<T, <V>(() => V) => V>): any
}
```

Eslint fails with error:

```
9:28  error  Parsing error: Unexpected token, expected )

   7 | type Props<T: Test> = {
   8 |   prop: T,
>  9 |   render(a: $ObjMap<T, <V>(() => V) => V>): any
     |                            ^
  10 | }
  11 |
```

**Comment by kentcdodds** [comment]
Thanks for reporting! Could you dig a little further and find out what the source of the issue is and maybe provide a test-case and potentially even a fix?

**Comment by michalkvasnicak** [comment]
@kentcdodds it seems to be problem with babylon

**Comment by michalkvasnicak** [comment]
Ok classic `rm -rf node_modules`, `yarn install` fixed a problem. Strange. Don't know what was causing this problem. I tried to simulate it using `babylon` and everything was fine.

Closing now.
